core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -334,6 +335,9 @@ private TableProperties() {}
   public static final String MAX_REF_AGE_MS = "history.expire.max-ref-age-ms";
   public static final long MAX_REF_AGE_MS_DEFAULT = Long.MAX_VALUE;
 
+  public static final String DELETE_GRANULARITY = "write.delete.granularity";
```

Comment:
Should we add a document for this table property?

Reply:
Will do, same for the write option.

Reply:
Added.

Reply:
Reverted based on [this](https://github.com/apache/iceberg/pull/9384/files#r1436606249) discussion.